.github/workflows/build-validation.yml

```diff
@@ -7,9 +7,15 @@ on:
   pull_request:
     branches: [ main ]
     types: [opened, synchronize, reopened]
+  workflow_dispatch:
+    inputs:
+      reason:
+        description: 'The reason for running the workflow'
+        required: true
+        default: 'Manual run'
```

Comment:
Was the workflow-dispatch addition intentional?

Reply:
Yes. Once this gets merged, you can run tests on your fork before opening a PR. 

(Notice that I had to make one additional change, because I missed one flag that needs to change)